app/models/concerns/foreman_remote_execution/orchestration/ssh.rb

```diff
@@ -0,0 +1,58 @@
+module ForemanRemoteExecution
+  module Orchestration::SSH
+    extend ActiveSupport::Concern
+
+    included do
+      before_destroy :ssh_destroy
+      after_validation :queue_ssh_destroy
+      register_rebuild(:queue_ssh_destroy, N_("SSH_#{self.to_s.split('::').first}"))
+    end
+
+    def drop_from_known_hosts(args)
+      proxy_id, target = args
+      proxy = ::SmartProxy.find(proxy_id)
+      begin
+        proxy.drop_host_from_known_hosts(target)
+      rescue => e
+        Rails.logger.warn e.message
+        return false
+      end
+      true
+    end
+
+    def ssh_destroy
+      logger.debug "Scheduling SSH known_hosts cleanup"
+
+      host, _kind, target = host_kind_target
+      proxies = %w(SSH Ansible).map { |kind| host.remote_execution_proxies(kind).values }
```

Comment:
Why do we care about Ansible here? it just sounds wrong :thinking: 
We can't have Ansible without SSH anyway, or am I missing something?
```suggestion
      proxies = host.remote_execution_proxies('SSH').values
```

Reply:
Right, we shouldn't have proxies with ansible without ssh. I just wanted to explicitly mention anything that could put records into the know hosts file

Reply:
hmm understood, it just feels weird as REX should ideally know nada about Ansible, right?

Reply:
You're right, changed.